### ISSUE TITLE
Creating a torrent from a directory with a pipe char in the name disrupts file verification

### ISSUE THREAD

**Issue by tomholford** [transmission/transmission#7123]
### What is the issue?

To reproduce:

1. Create a directory with pipe characters (e.g., `| TEST |`), and add some data to it
2. From the Transmission UI, click Create New Torrent and select the directory
3. Observe in the following modal dialog that the file shows 0.00% of x bytes; in other words, Transmission is not able verify the data:

<img width="478" alt="image" src="https://github.com/user-attachments/assets/714d74a9-c239-4583-8c0b-e4fd9917c525">

Normally, this will show 100% for each available local file.

4. Upon clicking Add, the torrent is added to the index view, and shows that it doesn't have any local data:

<img width="379" alt="image" src="https://github.com/user-attachments/assets/95b30d92-8cfd-4f8c-ae48-2d2a8b223e47">

Perhaps the pipe character is messing up the path resolution logic.

macOS 14.6.1 (23G93)
Transmission 4.0.6 (38c164933e)

### Which application of Transmission?

macOS app

### Which version of Transmission?

4.0.6 (38c164933e)

**Comment by nevack** [comment]
Cannot reproduce

<img width="1024" alt="image" src="https://github.com/user-attachments/assets/d4566800-99e3-4945-b734-ab2f32802d12">
<img width="1024" alt="image" src="https://github.com/user-attachments/assets/7c0e6424-e5d5-4928-9ca2-b649cb5cc013">
<img width="1024" alt="image" src="https://github.com/user-attachments/assets/c20f75d3-1b6e-43e3-b9fd-c908ba8305e0">

Are you sure you have selected correct destination directory?

**Comment by killemov** [comment]
No! Pipe characters are illegal to use as file/directory names.

**Comment by valcomm** [comment]
> No! Pipe characters are illegal to use as file/directory names.

Since when?  In linux you cannot use only '/' and null character.  Everything else as allowed and legal.

**Comment by tomholford** [comment]
> Cannot reproduce on main@[f6e1bb2](https://github.com/transmission/transmission/commit/f6e1bb2d700750266365da1fcc31b1af26bb18cd)

@nevack My mistake, I assumed it was the pipe char in my initial testing. For reference, here's the original string that I copy and pasted as the file name (from [this video](https://www.youtube.com/watch?v=TytFcQq6G-0), check it out if you like vaporwave):

```
|　𝙁𝙊𝙍 𝙇𝙀𝘼𝙎𝙀　|　Best of Mallsoft & Vaporwave Mix
```

Looks like the [`U+3000`](https://www.compart.com/en/unicode/U+3000) character used as white space between the pipes could be causing the issue. Was able to repro:

<img width="473" alt="image" src="https://github.com/user-attachments/assets/52eb535f-516e-4130-9120-65136968cfd5">

**Comment by nevack** [comment]
> @tomholford So close this issue?

Issue wan not correctly reported first time, but can confirm with updated info.

**Comment by tomholford** [comment]
> > @tomholford So close this issue?
> 
> Issue wan not correctly reported first time, but can confirm with updated info.

@nevack I have updated the ticket's name and description to reflect the actual issue; if a new ticket is necessary, I can also create one and close this instead, just lmk.
